packages/model/history.go

```diff
@@ -0,0 +1,49 @@
+package model
+
+import (
+	"errors"
+	"fmt"
+	"time"
+)
+
+const keysUpdatePattern = `UPDATE "1_keys" SET amount = amount + ? WHERE id = ?`
+
+var errLowBalance = errors.New("not enough APL on the balance")
+
+// History represent record of history table
+type History struct {
+	ID          int64
+	SenderID    int64
+	RecipientID int64
+	Amount      float64
+	Comment     string
+	BlockID     int64
+	TxHash      []byte
+	CreatedAt   time.Time
+}
+
+// TokenTransferWithHistory add one history record and update wallet balance
+func TokenTransferWithHistory(transaction *DbTransaction, history History) error {
+	db := GetDB(transaction)
+	r := db.Raw(`SELECT amount FROM "1_keys" WHERE id = ?`, history.SenderID).Row()
```

Comment:
1) use SELECT ... WHERE id = ? FOR UPDATE, to prevent race condition in that case
2) Use of raw queries is prohibited, except case that query too complex, and you cannot use gorm ORM or query builder for that query